config/config.go

```diff
@@ -446,6 +446,8 @@ type GlobalConfig struct {
 	// Keep no more than this many dropped targets per job.
 	// 0 means no limit.
 	KeepDroppedTargets uint `yaml:"keep_dropped_targets,omitempty"`
+	// Allow UTF8 Metric and Label Names
+	AllowUTF8Names bool `yaml:"utf8_names,omitempty"`
```

Comment:
Hm, wonder if "escaping scheme" is not a better config option? Ideally we avoid bools and inflexibility of it.

Reply:
so an escaping scheme would imply utf8 names being accepted? I think that would be ok

Reply:
Escaping schema is something different from allowing utf8. I was confused initially as well until I checked out `common/model`. See [this](https://github.com/prometheus/common/blob/a3eaa75cdb265e37302f56b41595ac4174b2c8e2/model/metric.go#L43-L77); allowing utf8 is a validation schema and has only two validation schemas (utf8 or legacy). I think we can keep it boolean for now.

Reply:
I just realised that this is a config that users set and will be a breaking change if we choose to change it. Maybe boolean indeed is not the best. How about we have it as `metric_name_validation_schema` with `legacy` and `utf8` as two options? The code later can choose to convert into boolean for using it for checks.

Reply:
> so an escaping scheme would imply utf8 names being accepted? I think that would be ok

I would rather be explicit and not change validation schema based on escaping schema at the moment.

Reply:
ok sounds good